src/machine/machine_rp2040-rtc.go

```diff
@@ -0,0 +1,256 @@
+	rtc.setup0.SetBits(uint32(t.Month) << rp.RTC_SETUP_0_MONTH_Pos)
+	rtc.setup0.SetBits(uint32(t.Day) << rp.RTC_SETUP_0_DAY_Pos)
+
+	rtc.setup1.SetBits(uint32(t.Dotw) << rp.RTC_SETUP_1_DOTW_Pos)
+	rtc.setup1.SetBits(uint32(t.Hour) << rp.RTC_SETUP_1_HOUR_Pos)
+	rtc.setup1.SetBits(uint32(t.Min) << rp.RTC_SETUP_1_MIN_Pos)
+	rtc.setup1.SetBits(uint32(t.Sec) << rp.RTC_SETUP_1_SEC_Pos)
+
+	// Load setup values into rtc clock domain
+	rtc.ctrl.SetBits(rp.RTC_CTRL_LOAD)
+
+	// Enable RTC and wait for it to be running
+	rtc.ctrl.SetBits(rp.RTC_CTRL_RTC_ENABLE)
+	for !rtc.running() {
+	}
+
+	return nil
+}
+
+func (rtc *rtcType) GetTime() (t RtcTime, err error) {
+	// Make sure RTC is running
+	if !rtc.running() {
+		return RtcTime{}, ErrRtcNotRunning
+	}
+
+	// Note: RTC_0 should be read before RTC_1
+	rtc_0 := rtc.rtc0.Get()
+	rtc_1 := rtc.rtc1.Get()
+
+	t = RtcTime{
+		Dotw:  int8((rtc_0 & rp.RTC_RTC_0_DOTW_Msk) >> rp.RTC_RTC_0_DOTW_Pos),
+		Hour:  int8((rtc_0 & rp.RTC_RTC_0_HOUR_Msk) >> rp.RTC_RTC_0_HOUR_Pos),
+		Min:   int8((rtc_0 & rp.RTC_RTC_0_MIN_Msk) >> rp.RTC_RTC_0_MIN_Pos),
+		Sec:   int8((rtc_0 & rp.RTC_RTC_0_SEC_Msk) >> rp.RTC_RTC_0_SEC_Pos),
+		Year:  int16((rtc_1 & rp.RTC_RTC_1_YEAR_Msk) >> rp.RTC_RTC_1_YEAR_Pos),
+		Month: int8((rtc_1 & rp.RTC_RTC_1_MONTH_Msk) >> rp.RTC_RTC_1_MONTH_Pos),
+		Day:   int8((rtc_1 & rp.RTC_RTC_1_DAY_Msk) >> rp.RTC_RTC_1_DAY_Pos),
+	}
+
+	return t, nil
+}
+
+// void rtc_set_alarm(datetime_t *t, rtc_callback_t user_callback) {
+func (rtc *rtcType) SetAlarm(t RtcTime, callback func()) {
+
+	rtc.disableInterruptMatch()
+
+	// Only add to setup if it isn't -1
+	// Set the match enable bits for things we care about
+
+	if t.Year >= 0 {
+		rtc.irqSetup0.SetBits(uint32(t.Year) << rp.RTC_SETUP_0_YEAR_Pos)
+		rtc.irqSetup0.SetBits(rp.RTC_IRQ_SETUP_0_YEAR_ENA)
+	}
+
+	if t.Month >= 0 {
+		rtc.irqSetup0.SetBits(uint32(t.Month) << rp.RTC_SETUP_0_MONTH_Pos)
+		rtc.irqSetup0.SetBits(rp.RTC_IRQ_SETUP_0_MONTH_ENA)
+	}
+
+	if t.Day >= 0 {
+		rtc.irqSetup0.SetBits(uint32(t.Day) << rp.RTC_SETUP_0_DAY_Pos)
+		rtc.irqSetup0.SetBits(rp.RTC_IRQ_SETUP_0_DAY_ENA)
+	}
+
+	if t.Dotw >= 0 {
+		rtc.irqSetup1.SetBits(uint32(t.Dotw) << rp.RTC_SETUP_1_DOTW_Pos)
+		rtc.irqSetup1.SetBits(rp.RTC_IRQ_SETUP_1_DOTW_ENA)
+	}
+
+	if t.Hour >= 0 {
+		rtc.irqSetup1.SetBits(uint32(t.Hour) << rp.RTC_SETUP_1_HOUR_Pos)
+		rtc.irqSetup1.SetBits(rp.RTC_IRQ_SETUP_1_HOUR_ENA)
+	}
+
+	if t.Min >= 0 {
+		rtc.irqSetup1.SetBits(uint32(t.Min) << rp.RTC_SETUP_1_MIN_Pos)
+		rtc.irqSetup1.SetBits(rp.RTC_IRQ_SETUP_1_MIN_ENA)
+	}
+
+	if t.Sec >= 0 {
+		rtc.irqSetup1.SetBits(uint32(t.Sec) << rp.RTC_SETUP_1_SEC_Pos)
+		rtc.irqSetup1.SetBits(rp.RTC_IRQ_SETUP_1_SEC_ENA)
+	}
+
+	// Does it repeat? I.e. do we not match on any of the bits
+	rtcAlarmRepeats = t.AlarmRepeats()
+
+	// Store function pointer we can call later
+	rtcCallback = callback
+
+	// Enable the IRQ at the proc
+	interrupt.New(rp.IRQ_RTC_IRQ, rtcHandleInterrupt).Enable()
+	irqSet(rp.IRQ_RTC_IRQ, true)
```

Comment:
Is this really needed?
I see this pattern of `interrupt.New().Enable()` followed by `irqSet()` across rp2040 related code.
Yet, it works w/o `irqSet()` call too.

Reply:
@aykevl what would be your understanding?
If this `irqSet` is indeed not needed in such cases, I could make a PR that cleans this in RP2040 machine.